### PR TITLE
108: added support for value arrays when working with objects

### LINKED ISSUE #108: Value array support
Starting from the following example, I would like to simplify the `query2` code like the `query` is.

The `query` output leads to an `"[object object]"` string replacement, I think it would be good to support an "auto-glue" option to avoid the `SQL.glue` statement for these simple tasks.

```
const SQL = require('@nearform/sql')

const ids = [1, 2, 3]

const query = SQL`SELECT * FROM table WHERE id IN (${ids})`
console.log(query.sql)
console.log(query.values)
// SELECT * FROM table WHERE id IN (?)
// [ [ 1, 2, 3 ] ]

const query2 = SQL`SELECT * FROM table WHERE id IN (${SQL.glue(
  ids.map((id) => SQL`${id}`),
  ','
)})`
console.log(query2.sql)
console.log(query2.values)
// SELECT * FROM table WHERE id IN (?,?,?)
// [ 1, 2, 3 ]

```

### PR DESCRIPTION
**Pull request by lasmil**
- when working with arrays of objects, we should be able to create a clean SQL query using glue
- example of usage can be found in the SQL.test.js file
- closes #108

### CODE REVIEW

File: SQL.js [hunk 1]
```diff
@@ -35,6 +35,20 @@ class SqlStatement {
     return new SqlStatement(result.strings, result.values)
   }
 
+  /**
+   * A function that accepts an array of objects and a mapper function
+   * It returns a clean SQL format using the object properties defined in the mapper function
+   */
+  map (array, mapFunc) {
+    if ((mapFunc instanceof Function) && array && array.length > 0) {
```
Comment (ilteoood) [inline SQL.js:43]:
```suggestion
    if ((mapFunc instanceof Function) && array?.length) {
```

File: SQL.js [hunk 2]
```diff
@@ -35,6 +35,20 @@ class SqlStatement {
...
+      return this.glue(
+        array.map(mapFunc).map((item) => SQL`${item}`),
```
Comment (ilteoood) [inline SQL.js:45]: as SQL should be a function, technically you can reduce it to:
```suggestion
        array.map(mapFunc).map(SQL),
```

File: SQL.js (hunk 1 above)
Comment (lasmil) [reply SQL.js:43]: Yeah, I am getting Unexpected token '.' most probably because of the version of node that I am using

Comment (lasmil) [reply SQL.js:43]: I have updated the node version and now it works

File: SQL.js (hunk 2 above)
Comment (lasmil) [reply SQL.js:45]: I tried it and the tests failed. I got undefined values instead of the actual IDs

File: SQL.js [hunk 3]
```diff
@@ -35,6 +35,20 @@ class SqlStatement {
     return new SqlStatement(result.strings, result.values)
   }
 
+  /**
+   * A function that accepts an array of objects and a mapper function
+   * It returns a clean SQL format using the object properties defined in the mapper function
+   */
+  map (array, mapFunc) {
```
Comment (Eomm) [inline SQL.js:42]: I wonder if a default mapFunc equals to `i => i` could be even more easy to use.

Here a sample usage without the default:

```
// as is now
const query2 = SQL`SELECT * FROM table WHERE id IN (${SQL.map(ids, i=>i)})`

// with default:
const query2 = SQL`SELECT * FROM table WHERE id IN (${SQL.map(ids)})`
```

**Review by Eomm — COMMENTED** [review]
The feature LGTM, I added just a note

I think the readme needs to be updated as well

File: SQL.js (hunk 3 above)
Comment (lasmil) [reply SQL.js]: Done, I have added a default mapper function

File: README.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ A simple SQL injection protection module that allows you to use ES6 template str
    1. [Linting](#linting)
 3. [Methods](#methods)
    1. [glue](#gluepieces-separator)
+   2. [map](#map)
```
Comment (simoneb) [inline README.md:12]: are you sure that this anchor link works? because the title name is different from the anchor name

File: README.md [hunk 5]
```diff
@@ -117,6 +118,40 @@ const sql = SQL`INSERT INTO users (id, name) VALUES
...
+  name: 'name2'
+},
+{
+  id: 3,
+  name: 'name3'
+}]
+
+const mapperFunction = (objItem) => {
+  return objItem.id
+}
```
Comment (simoneb) [inline README.md:149]: why don't you put this inline? I think it would be a more common use case

File: README.md (hunk 4 above)
Comment (lasmil) [reply README.md]: I found the problem, pushed to PR.

File: README.md (hunk 5 above)
Comment (lasmil) [reply README.md]: Yepp, and it's easier to read. Pushed a commit.

**Pull request merged**